app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -1,24 +1,32 @@
 package org.schabi.newpipe.fragments.detail;
 
+import android.animation.ValueAnimator;
```

Comment:
I can't agree with this idea because there is not much space in the top line.
Seven buttons (captions, playlist, speed, quality, kodi, share, open in browser) is too much even if we make captions as a button with icon instead of text. It's too much as a number of buttons and as a wasted space. 
To be honest I never use buttons under hidable menu. OK, maybe once in a month I can share or open in browser. That's it. It doesn't make sense to make them always visible. I don't want to see them every time I tap on a screen. 
I never, never used a "fit" to screen button BUT maybe i'm not that guy who need it. Maybe this button is useful for those who has non standart screen aspect ratio. Maybe it's must have for those people, we should ask them first before removing this button.  
So many buttons in a row is too distracting. You are talking about number of taps to get the desired action. It's important, I agree. And the question is do you actually need to be two taps away from getting to captions button? Or to Kodi button? I don't think so. That's why such hidden menu is important.

As a solution for a "number of taps" problem I can offer MX player variant when you can select menu under three dots icon and without releasing a finger you can move around the menu and select a submenu this way (captions can have submenu).

Reply:
Maybe the `360p` and `1x` buttons can be moved, respectively, left and right of the progress bar?

Reply:
> In the current version (0.18.0) it is possible to change the video resolution and share the video via the toolbar directly with one tap.

Oh no. This is a regression I noticed just now. I think those buttons are necessary. Being able to select the resolution before starting the video is important.

And not only is having to play the video to share it tedious, it also means the user only has the option of sharing a timestamped URL, which they may have to edit before using.

Maybe the top bar should be kept as it is. I don't see the harm. Or, it could be implemented below the video? But that seems far more complex.

Reply:
The problem with top bar is you can put popup mode at top and view comments easily at the same time.

Reply:
I didn't get it. You can't see the comments at the same time currently, either.